### PR TITLE
Return max heap size note

### PR DESCRIPTION
**Pull request by afwn90cj93201nixr2e1re**
Fixes: #831

With note:
изображение

Without note:
изображение

### CODE REVIEW

File: compiler/libpc300/sc1.c [hunk 1]
```diff
@@ -140,7 +140,7 @@ static int nestlevel  = 0;      /* number of active (open) compound statements *
 static int rettype    = 0;      /* the type that a "return" expression should have */
 static int skipinput  = 0;      /* number of lines to skip from the first input file */
 static int optproccall = TRUE;  /* support "procedure call" */
-static int verbosity  = 1;      /* verbosity level, 0=quiet, 1=normal, 2=verbose */
+static int verbosity  = 3;      /* verbosity level, 0=quiet, 1=normal, 2=verbose */
```
Comment (afwn90cj93201nixr2e1re) [inline compiler/libpc300/sc1.c]: Oops.

**Comment by afwn90cj93201nixr2e1re** [comment]
thanks for fast response.

**Comment by afwn90cj93201nixr2e1re** [comment]
@Arkshine same here. Im using it since release, so, there's no bug's, just merge it...

**Comment by Arkshine** [comment]
I believe this code has been removed for a good reason. I can't remember why. It's probably somewhere in some PR related to compiler updates.

**Comment by afwn90cj93201nixr2e1re** [comment]
No, u just removed it without any reasons.
This code also still exists in other pawn project's and it's works well.

**Comment by afwn90cj93201nixr2e1re** [comment]
Is that still present in sp?  Any ideas? It's really not hurting anything, not breaking anything, just notification, coz sometimes even I can't manually calc., or imagine how many cells were used by the compiler in order to understand whether I am close to the limit or not, etc.

@dvander

**Comment by asherkin** [comment]
> I believe this code has been removed for a good reason. I can't remember why. It's probably somewhere in some PR related to compiler updates.

> No, u just removed it without any reasons.

As Arkshine predicted, the PR that removed this says exactly why it was removed: https://github.com/alliedmodders/amxmodx/pull/137

This change seems to regress that again, and the information isn't very valuable either in my opinion.

> Is that still present in sp?

No.

**Comment by afwn90cj93201nixr2e1re** [comment]
So his real suggestion was about recursion only, and ofc for less knowledge ppls which are just spamming on forums and asking why there's warning and how to fix it? So, maybe then we should just remove recursion warnings? Or refactor default plugin code if they are spamming on compilation step?

> Because of recursion detection improvement added in previous patch, some plugins are spammed with recursion warnings even though usage is legit. To avoid to be spammed on forum as well because of this warning (like indentation one) and because this is actually not that useful, let's disable this.

So, it's exactly what shouldn't be done.
Maybe then we should remove 217 (tbh it's really useful warning due spaces/tabs with code without pragma semicolon, there's can be unexpected results too (on conditions)), 203 warning's too? How much topic's about these warning's was at forum?

I don't understand why the development of the kernel should depend on the opinion of less knowledge ppls which are doesn't know what does recursion mean and how they can avoid it in simple cases (for example: http://tristanpenman.com/blog/posts/2018/12/08/tail-call-optimisation-and-c-cpp/ and etc), they are just visiting forum and roar, troll, mock knowledgeable developers (often this went unpunished) instead of using the search, all of their questions already described in responses.

Let's be realistic, Arkshine removed this warning, how did this affect to coders? Nohow, he did not teach them how write right code logic, he only added even more unexpected results, which sometimes will cause more unexpected logic, the problem hasn't been solved, no.

> information isn't very valuable either in my opinion

For us? Ofc, coz we are know how it's works and **sometimes** can detect it visually (sometimes), but we aren't calculator's/compiler's and don't wanna unpack amxx to amx and check in hex/pawn disassembler how much space was used for specific function.
But sometimes there's code like that, im pretty sure that u aren't expect some stack corruption on some code's too, but they are also contain it.

```
public formatSomeMotdFromJsonString(id) {
	new jsonParts[1000][3]; // there we go | or you can replace it with some code, which also gonna take too much cells
	log_amx("%s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
}

public formatSomeMotdFromJsonString(id) {
	new jsonParts[7][545]; // there we go | or you can replace it with some code, which also gonna take too much cells
	// some parsing stuff on first
	// some parsing stuff on second
	// some parsing stuff on third
	// some parsing stuff on fourth
	// some parsing stuff on five
	// some parsing stuff on six
	// some parsing stuff on seven
	// all of parsing stuff also take some space

	log_amx("%s %s %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Second part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Third part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Fourth part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Five part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Sixth  part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));
	log_amx("Seventh part %s %s  %s %s", jsonParts[0], jsonParts[1], jsonParts[2], "another string", fmt("One more string %d", 1));

	log_amx("All strings together %s_%s_%s_%s_%s_%s", jsonParts[0], jsonParts[1], jsonParts[2], jsonParts[3], jsonParts[4], jsonParts[5], jsonParts[6]);
	// making some menu from json result
	new g_MainMenu1; new cb = menu_makecallback("Test");
	g_MainMenu1 = menu_create("TITLE", "MainH", true);
	menu_additem(g_MainMenu1, "TEST");
	menu_additem(g_MainMenu1, "TEST", .paccess = 123);
	for(new i = 0, SomeSize = 7; i < SomeSize; i++) {
		menu_additem(g_MainMenu1, fmt("some string %s", jsonParts[0][i]));
		menu_addblank2(g_MainMenu1);
	}
	menu_additem(g_MainMenu1, "ENABLE_DISABLE", .callback = cb); // 6 - 1
	menu_setprop(g_MainMenu1, MPROP_PAGE_CALLBACK, "MakeMainMenuExitMultilangual");

	new g_MainMenu2AnotherVariableName = menu_create("TITLE2", "MainH2", true);
	menu_additem(g_MainMenu2AnotherVariableName, "TEST");
	menu_additem(g_MainMenu2AnotherVariableName, "TEST", .paccess = 123);
	for(new i = 0, SomeSize = 7; i < SomeSize; i++) {
		menu_additem(g_MainMenu2AnotherVariableName, fmt("some string %s", jsonParts[0][i]));
		menu_addblank2(g_MainMenu2AnotherVariableName);
	}
	menu_additem(g_MainMenu2AnotherVariableName, "ENABLE_DISABLE", .callback = cb); // 6 - 1
	menu_setprop(g_MainMenu2AnotherVariableName, MPROP_PAGE_CALLBACK, "MakeMainMenuExitMultilangual");
}
```

We can understand that there's possible stack corruption due too much cells usage (just refresh on ur mind that's amx pawn, not sourcepawn where we can do what we wanna and optimize/add new compiler/instructions and etc), and gonna use static or global new (instead stack + data)/split code and etc to reduce stack usage, let's not depend on variables, their size, as i said in code comments, sometimes there's unexpected code which also generate 4kb+ usage at stack lvl and there's no notification about it.

If you really objectively think that these are useless warnings, then no problem, ok, I will close the PR, but as for me they are useful if you evaluate the situation objectively, not from our pov, but from the pov of an ordinary scipter's (from all sides) without such knowledge in asm/stack usage/c/cpp and etc.

tbh, I really do not understand why I am explaining all of this stuff, you should perfectly understand the situation from all sides, not only from your pov, but also from the pov of stupid people who code anyhow, and then come to the forum (let's be real, it's causing more topics then compiler notifications) and ask what I have here and why such results, and the bunch of the same people stretch the solution of the topic for months until someone comes and fixes the usual stack error.

This text narration, not rage, without unnecessary emotions, just explaining.

**Comment by dvander** [comment]
Please stop this. You're at risk of getting yourself banned from this project. This is your final warning.

The PR was committed *six years ago* to solve a real problem. Reverting it will re-introduce that problem. This is not a conspiracy against you, and does not warrant attacking anyone involved in the patch. If you want to address your issue, you need to find a way to do it that does not re-introduce old bugs.

My suggestion: automatically bump up the final stack/heap size based on the largest function stack frame. I think that'd be a 5-line patch.

**Comment by afwn90cj93201nixr2e1re** [comment]
> This text narration, not rage, without unnecessary emotions, just explaining.
> You're at risk of getting yourself banned from this project. This is your final warning.

whatever? Ok.

> The PR was committed six years ago to solve a real problem

Im not talking about PR, only about specific piece of this PR.
> Reverting it will re-introduce that problem

Im asking to revert it, you should take a closer look too.

> you need to find a way to do it that does not re-introduce old bugs.

How notification return can re-introduce bug?

> automatically bump up the final stack/heap size based on the largest function stack frame. I think that'd be a 5-line patch.

yes it is.

**Comment by afwn90cj93201nixr2e1re** [comment]
oh, and i forgot about this one in last email
https://github.com/alliedmodders/amxmodx/issues/696

just for you take into account that I am not the only one who feels the problem, in ru community we just rofling if someone commiting something to amxmodx, coz there's 0.000001% chances that it's gonna be merged.
изображение

> You're at risk of getting yourself banned from this project. This is your final warning.

Thanks for pointing it before banning, it gave me time.

Im not Karen, If you really can't distinguish between an attack and an conversation, which in no way implies a desire to  offend/accuse/humiliate someone, you can ban, since your warnings are already the initial stage, and I do not want to just in the middle of the next conversation get a banned because it seems to you that I'm attacking someone again.

**Pull request closed without merging**